stdlib/public/core/Substring.swift.gyb

```diff
@@ -317,42 +317,25 @@ extension Substring : LosslessStringConvertible {
   }
 }
 
-extension Substring : Equatable {
-  public static func ==(lhs: Substring, rhs: Substring) -> Bool {
-    return String(lhs) == String(rhs)
+extension StringProtocol {
+  public static func ==<S: StringProtocol>(lhs: Self, rhs: S) -> Bool {
```

Comment:
`R` or `RHS` would probably be a better generic parameter name.  There are enough `S`'s flying around here already